std/algorithm/searching.d

```diff
@@ -2849,7 +2849,28 @@ true if the initial segment of $(D r1) matches $(D r2), and $(D r1) has been
 advanced to the point past this segment; otherwise false, and $(D r1) is left
 in its original position.
  */
-bool skipOver(alias pred = "a == b", R1, R2)(ref R1 r1, R2 r2)
+bool skipOver(R1, R2)(ref R1 r1, R2 r2)
+    if (isForwardRange!R1 && isInputRange!R2
+        && is(typeof(r1.front == r2.front)))
+{
+    static if (isSomeString!R1 && isSomeString!R2 &&
+        r1[0].sizeof == r2[0].sizeof)
+    {
+        if (r2.length > r1.length || r1[0 .. r2.length] != r2)
```

Comment:
The other, non-string version of the function doesn't have this length check fast path. Could you add it for when both ranges have lengths?